x/chat/types/messages.go

```diff
@@ -0,0 +1,233 @@
+package types
+
+import (
+	types "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	proto "github.com/gogo/protobuf/proto"
+)
+
+// Chat message types
+const (
+	TypeMsgCreateChannel = "create_channel"
+	TypeMsgSendMessage   = "send_message"
+	TypeMsgVotePoll      = "vote_poll"
+)
+
+// Verify interface at compile time
+var (
+	_ sdk.Msg = &MsgCreateChannel{}
```

Comment:
not mandatory but these checks can be moved to tests

Reply:
I followed the Cosmos SDK implementation, I think the advantage is the the IDE can tell you the unimplemented method of your object in real time